code/modules/pool/pool_controller.dm

```diff
@@ -0,0 +1,330 @@
+//Originally stolen from paradise. Credits to tigercat2000.
+//Modified a lot by Kokojo and Tortellini Tony for hippiestation.
+//Heavily refactored by tgstation
+/obj/machinery/pool
+	icon = 'icons/obj/machines/pool.dmi'
+
+/obj/machinery/pool/controller
+	name = "\improper Pool Controller"
+	desc = "An advanced substance generation and fluid tank management system that can refill the contents of a pool to a completely different substance in minutes."
+	icon_state = "poolc_3"
+	anchored = TRUE
+	density = TRUE
+	use_power = TRUE
+	idle_power_usage = 75
+	var/list/linkedturfs //List contains all of the linked pool turfs to this controller, assignment happens on initialize
+	var/list/mobs_in_pool = list()//List contains all the mobs currently in the pool.
+	var/temperature = POOL_NORMAL //1-5 Frigid Cool Normal Warm Scalding
+	var/srange = 6 //The range of the search for pool turfs, change this for bigger or smaller pools.
+	var/list/linkedmist = list() //Used to keep track of created mist
+	var/misted = FALSE //Used to check for mist.
+	var/cur_reagent = "water"
+	var/drainable = FALSE
+	var/drained = FALSE
+	var/bloody = 0
+	var/obj/machinery/pool/drain/linked_drain = null
+	var/obj/machinery/pool/filter/linked_filter = null
+	var/interact_delay = 0 //cooldown on messing with settings
+	var/reagent_delay = 0 //cooldown on reagent ticking
+	var/shocked = FALSE//Shocks morons, like an airlock.
+	var/tempunlocked = FALSE
+	var/old_rcolor
+	resistance_flags = INDESTRUCTIBLE|UNACIDABLE
```

Comment:
this is lazy, these things should be constructable, it's the same thing with making the pool unexplodable, they're just too lazy to make it work properly.

Reply:
I'll add it to the todo list